gomanifest/build.go

```diff
@@ -0,0 +1,234 @@
+	Packages []MainfestDirectDeps `json:"packages"`
+}
+
+// Deps package from `go list -deps -json ./...` command
+var depsPackages = make(map[string]GoPackage)
+
+// transformVersion ... Converts the golang version string into semver without 'v' and appended text after '+'
+func transformVersion(inVersion string) string {
+	var outVersion string = strings.Replace(inVersion, "v", "", 1)
+
+	return strings.Split(outVersion, "+")[0]
+}
+
+// getPackageName ... Utility function to convert package + module data into package name used by manifest.
+func getPackageName(depPackage GoPackage) string {
+	// Get module / package@module
+	if depPackage.ImportPath != depPackage.Module.Path {
+		return depPackage.ImportPath + "@" + depPackage.Module.Path
+	}
+
+	return depPackage.ImportPath
+}
+
+// GoListCmd ... Go list command structure.
+type GoListCmd struct {
+	CWD string
+}
+
+// Run ... Actual function that executes go list command and returns output as string.
+func (goListCmd GoListCmd) Run() (string, error) {
+	GoListGoListDeps := exec.Command("go", "list", "-json", "-deps", "./...")
+	GoListGoListDeps.Dir = goListCmd.CWD
+	output, err := GoListGoListDeps.Output()
+
+	if err != nil {
+		return "", err
+	}
+
+	fmt.Println("Outp", string(output))
+	return string(output), nil
+}
+
+// GoListCmdInterface ... Interface to be implemented to execute go list command.
+type GoListCmdInterface interface {
+	Run() (string, error)
+}
+
+// GoList ... Structure that handle go list data and extract required packages.
+type GoList struct {
+	Command GoListCmdInterface
+}
+
+// Get ... Get deps data through go list deps command and converts json into objects.
+func (goList *GoList) Get() int {
+	output, err := goList.Command.Run()
+
+	if err != nil {
+		log.Println("ERROR :: Command `go list -json -deps ./...` failed, resolve project build errors.", err)
+		return -1
+	}
+
+	goListDepsData := string(output)
+	goListDepsData = `{"Packages": [` + strings.ReplaceAll(goListDepsData, "}\n{", "},\n{") + "]}"
+
+	var goPackages GoPackages
+	json.Unmarshal([]byte(goListDepsData), &goPackages)
+	log.Println("Packages in deps:", len(goPackages.Packages))
+
+	// Preprocess and remove all standard packages.
+	for i := 0; i < len(goPackages.Packages); i++ {
+		// Exclude standard packages
+		if goPackages.Packages[i].Standard == false {
+			depsPackages[goPackages.Packages[i].ImportPath] = goPackages.Packages[i]
+		}
+	}
+	log.Println("Filter package count:", len(depsPackages))
+
+	return 0
+}
+
+// getTransitives ... Returns a clean list of deps
+func getTransitives(deps []string) []ManifestTransitiveDeps {
+	var manifestDependencies = make([]ManifestTransitiveDeps, 0)
+	for i := 0; i < len(deps); i++ {
+		if depPackage, ok := depsPackages[deps[i]]; ok {
+			if depPackage.Module.Main == false {
+				manifestDependencies = append(manifestDependencies, ManifestTransitiveDeps{
+					getPackageName(depPackage),
+					transformVersion(depPackage.Module.Version),
+				})
```

Comment:
Can this be abstracted in a function something like `NewDep` which takes `depPackage` as an arg and constructs ManifestTransitiveDeps?
```suggestion
				manifestDependencies = append(manifestDependencies, NewDep(depPackage))
```

Reply:
This is done for direct deps and transitive deps.